content/resources/operator/operator.md

```diff
@@ -0,0 +1,66 @@
+2. docker version 17.03+
+3. kubectl version v1.11.3+
+4. Access to a Kubernetes v1.11.3+ cluster ([kind](https://kind.sigs.k8s.io/ "kind") or [minikube](https://minikube.sigs.k8s.io/docs/ "minikube") are good enough)
+5. kubebuilder ([installation](https://book.kubebuilder.io/quick-start.html#installation "installation"))
+
+
+## Creating an operator
+
+Useful links:
+1. The [official documentation](https://book.kubebuilder.io/quick-start.html) is rather good but the example might appear far from what you want to realize.
+2. This [step-by-step tutorial](https://www.youtube.com/watch?v=KBTXBUVNF2I) on how to build an operator from scratch, after a first watch to get an overview, it might be useful to come back to it while you build the operator to look for specific parts.
+
+Useful information to get the kubebuilder project started:
+- Use `polito.it` as domain when doing `kubebuilder init` (or `example.com`  if you are just testing stuff)
+- The api group should be set to `crownlabs`
+- At the moment of writing CrownLabs is focused on developing the API version `v1alpha2`. Check out the repository to make sure you create your project with the right parameters.
+- the `kind` option is going to be the CRD name, the generated controller will be named `<kind_name>_controller`
+
+Sometimes you might want to create an operator that manages different kind of resources that have already been defined or simply you do not need to create a new CRD.
+Kubebuilder will generate the controller code only after generating CRDs, so even if you do not really need to define a CRD just do it and you can easily get rid of it later while you have the rest of the code correctly generated.
+
+
+## Writing the controller or modifying an existing operator
+
+The kubebuilder framework uses the controller-runtime library to communicate with the Kubernetes API. The documentation of the library is sparse across different packages, here are some useful ones:
+1. [controller-runtime](https://godoc.org/sigs.k8s.io/controller-runtime "controller-runtime") general godoc page
+2. controller-runtime\'s [client](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client "client") used to interact with the k8s API
+3. the [buider](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/builder "buider") used in the `SetupWithManager` function to select which resources to watch
+4. the [request](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Request "request") received in the reconciler function
+5. k8s [corev1](https://godoc.org/k8s.io/api/core/v1 "corev1") in case you need to interact with core k8s kind (pods, deployments, configmaps etc)
+
+
+## Notes
+
+You might notice that the structure in the CrownLabs repository is different from the one generated by kubebuilder. However, this is not a big deal:
+- `main.go` goes inside its own folder inside `operators/cmd/<operator-name>`
+- `<kind>_controller.go` goes in `operators/pkg/controllers/<kind>_controller.go`
```

Comment:
IIUC, the idea is to put each controller (or related controllers) in its own directory (instead of having a global controllers folder). Am I correct @palexster?

Reply:
Totally agree. I suggest to split the pkgs in different directories.

Reply:
Ok, I just saw the [PR#365](https://github.com/netgroup-polito/CrownLabs/pull/365).
If I understood correctly, it should be like this:
```suggestion
- `<kind>_controller.go` goes inside its own directory `operators/pkg/<kind>-controller/`
```

Reply:
Yes, thanks!